src/GusApi/Client/GusApiClient.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace GusApi\Client;
+
+use GusApi\Context\ContextInterface;
+use GusApi\Exception\NotFoundException;
+use GusApi\RegonConstantsInterface;
+use GusApi\Type\GetFullReport;
+use GusApi\Type\GetFullReportResponse;
+use GusApi\Type\GetValue;
+use GusApi\Type\GetValueResponse;
+use GusApi\Type\Login;
+use GusApi\Type\LoginResponse;
+use GusApi\Type\Logout;
+use GusApi\Type\LogoutResponse;
+use GusApi\Type\SearchData;
+use GusApi\Type\SearchDataResponse;
+use GusApi\Type\SearchResponseRaw;
+use GusApi\Util\DataSearchDecoder;
+use GusApi\Util\FullReportResponseDecoder;
+
+class GusApiClient
```

Comment:
Poprzednio korzystałeś tutaj z wzorca Adapter. Myślę, że nie warto z tego rezygnować, ponieważ GUS udostępnia też RESTowe API, ktore mogłoby być alternatywą dla SOAP. W przyszłości można też dodać drugą iplementację, dla osób, które nie chcą instalować rozszerzenia `soap`.

Reply:
Nic mi nie wiadomo o api rest do tej usługi, rzucisz jakiś dokument ?

Reply:
Niestety żadnej dokumentacji nie znalazłem, ale dowiedziałem się o tym z Twojego postu: https://4programmers.net/Forum/PHP/244462-wyciaganie_danych_z_gusu_nowa_wersja_?p=1100806#id1100806

Wygląda na to, że endpointy nadal działają, ale trzeba to rozpracować metodą prób i błędów :smiley: